pallets/author-reward-dest/src/lib.rs

```diff
@@ -0,0 +1,181 @@
+//! Pallet for setting and getting author reward destination.
+//!
+//! The author of a block can provide an account using inherents, this account is meant to be used
+//! for reward within the block such as tips and part of transaction fees.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use primitives_author_reward_dest::{InherentError, InherentType, INHERENT_IDENTIFIER};
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The type `Self::AccountId` but with additional constraint.
+		type AccountIdType: IsType<<Self as frame_system::Config>::AccountId> + From<InherentType>;
```

Comment:
beautiful example of https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/frame_runtime_types/index.html#adding-further-constraints-to-runtime-composite-enums